backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala

```diff
@@ -157,24 +163,45 @@ object BackendSettings extends BackendSettingsApi {
       }.headOption
     }
 
-    def validateFieldMetadata(fields: Array[StructField]): Option[String] = {
+    def validateFieldMetadata(): Option[String] = {
       if (fields.exists(!_.metadata.equals(Metadata.empty))) {
         Some("StructField contain the metadata information.")
       } else None
     }
 
-    def validateFileFormat(format: FileFormat): Option[String] = {
+    def validateFileFormat(): Option[String] = {
       format match {
         case _: ParquetFileFormat => None
         case _: FileFormat => Some("Only parquet fileformat is supported in native write.")
       }
     }
 
-    val fileFormat = validateFileFormat(format)
-    val metadata = validateFieldMetadata(fields)
-    val dataTypes = validateDateTypes(fields)
-    val compression = validateCompressionCodec()
-    compression.orElse(dataTypes).orElse(metadata).orElse(fileFormat)
+    def validateWriteFilesOptions(): Option[String] = {
+      val maxRecordsPerFile = options
+        .get("maxRecordsPerFile")
+        .map(_.toLong)
+        .getOrElse(SQLConf.get.maxRecordsPerFile)
+      if (maxRecordsPerFile > 0) {
+        Some("Unsupported native write: maxRecordsPerFile not supported.")
+      } else {
+        None
+      }
+    }
+
+    def validateBucketSpec(): Option[String] = {
+      if (bucketSpec.nonEmpty) {
+        Some("Unsupported native write: bucket write is not supported.")
+      } else {
+        None
+      }
+    }
+
+    validateCompressionCodec()
+      .orElse(validateFileFormat())
+      .orElse(validateFieldMetadata())
+      .orElse(validateDateTypes(fields))
```

Comment:
Can we also remove the fields parameters in validateDateTypes method?

Reply:
sure, removed